crates/stc_ts_file_analyzer/src/analyzer/assign/tpl.rs

```diff
@@ -74,6 +82,49 @@ impl Analyzer<'_, '_> {
                     return Ok(true);
                 }
 
+                if target.is_bool() {
+                    let value = &*value.value;
+
+                    match value {
+                        "true" | "false" => {
+                            return Ok(true);
+                        }
+                        "TRUE" | "FALSE" if !data.dejavu.is_empty() => {
+                            if let Type::Instance(ty) = &data.dejavu[0].0.normalize() {
```

Comment:
No idea if relying on `data.dejavu` here is problematic or not but I wanted to consult you before trying anything else since I think other solutions will be significantly harder

Reply:
You should not use `data.dejavu`